packages/workflow-core/src/lib/utils/deep-merge-with-options.ts

```diff
@@ -0,0 +1,139 @@
+import { isObject } from '@ballerine/common';
+
+type UnknownRecord = Record<PropertyKey, unknown>;
+
+const mergeObjects = (obj1: UnknownRecord, obj2: UnknownRecord) => {
+  const result = { ...obj1 };
+
+  for (let key in obj2) {
+    if (isObject(obj2[key]) && !Array.isArray(obj2[key]) && key in result) {
+      result[key] = mergeObjects(result[key] as UnknownRecord, obj2[key] as UnknownRecord);
+    } else {
+      result[key] = obj2[key];
+    }
+  }
+
+  return result;
+};
```

Comment:
Do you might need deep copy in order to prevent dealing with shallow copy?

Reply:
I don't want to touch any of this logic. It's an existing logic, it works and I don't have time to refactor it now. Sorry, and thanks for the comments